### PR TITLE
Refactor classes to use hooks

### PR DESCRIPTION
**Pull request by krisbobo**
## Milestone: Hooks
*The following changes are being introduced:*
- Refactor code to use hooks
- Code doesn't run on the browser without errors. A new entry prop is introduced. All efforts to resolve this have failed. An alternative look into it is needed.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,60 +1,54 @@
-import React from 'react';
+import { useState } from 'react';
 import calculate from '../logic/calculate';
```
Comment (HAMMAS-SALEEM) [inline src/components/Calculator.js:3]: Capture

- [ ] When i launched your application and clicked on number 8 then i saw **undefined** on your screen 😨 which you can see in the image above 🤔 Kindly fix it 👍

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @krisbobo :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (krisbobo) [reply src/components/Calculator.js:3]: Hi @HAMMAS-SALEEM 
Thanks for pointing that out. Any tips on how I can resolve that?

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by krisbobo** [comment]
Thanks @CollinsTatang

**Pull request merged**